### PR TITLE
fix(Scripts/SSC): Update Hydross Mark Timers

### PR DESCRIPTION
**Pull request by Alethrion**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/6451

## SOURCE:
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
https://www.icy-veins.com/tbc-classic/hydross-the-unstable-guide-strategy-abilities-loot
https://youtu.be/FEYAX7l5H6A?t=646
*Sound warning* https://youtu.be/J1Vph9r5ZnY&t=47
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks 

## Tests Performed:
This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [X] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:
- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [X] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Go to SSC's Hydross the Unstable
2. Move him across the flags
3. Check that there are 15s before he applies either Mark of Hydross or Mark of Corruption

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by elthehablo** [comment]
This does seem to roughly line up with sniffs. The repeating part is correct indeed. The highest level mark should be recast. Preferably would like to get a confirmation on the init timer though, as it seems slightly higher in the sniff I have.

**Comment by elthehablo** [comment]
TC also has 15 seconds init and 15 seconds repeating

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_hydross_the_unstable.cpp [hunk 1]
```diff
@@ -145,7 +145,7 @@ struct boss_hydross_the_unstable : public BossAI
             me->ApplySpellImmune(0, IMMUNITY_SCHOOL, SPELL_SCHOOL_MASK_NATURE, true);
             DoCastSelf(SPELL_CORRUPTION, true);
 
-            scheduler.Schedule(0s, GROUP_ABILITIES, [this](TaskContext)
+            scheduler.Schedule(15s, GROUP_ABILITIES, [this](TaskContext)
             {
                 DoCastSelf(SPELL_MARK_OF_CORRUPTION1);
             }).Schedule(15s, GROUP_ABILITIES, [this](TaskContext)
```
Comment (elthehablo) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_hydross_the_unstable.cpp:151]:
```suggestion
            }).Schedule(30s, GROUP_ABILITIES, [this](TaskContext)
```

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_hydross_the_unstable.cpp [hunk 2]
```diff
@@ -160,9 +160,10 @@ struct boss_hydross_the_unstable : public BossAI
             }).Schedule(60s, GROUP_ABILITIES, [this](TaskContext)
```
Comment (elthehablo) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_hydross_the_unstable.cpp:160]: The rest also need to be incremented by 15 seconds. With the current state both the first and second mark are cast at the same time

**Review by elthehablo — CHANGES_REQUESTED** [review]

**Comment by Alethrion** [comment]
Sound warning https://youtu.be/J1Vph9r5ZnY&t=47
This video is at 2x speed and it takes 7.5 seconds (47s-55s) for the first hydross mark to be applied, but I would not say no to redundant proof or proof to the contrary.

**Comment by elthehablo** [comment]
This is fine to merge for now. If it lines up with TC. I would need someone to go in on retail and specifically test the mark timers to get sniffed timers

**Review by elthehablo — APPROVED** [review]

**Pull request merged**